src/Compilers/CSharp/Test/Semantic/Semantics/PatternMatchingTests3.cs

```diff
@@ -2281,5 +2281,723 @@ public class D { }
+        {
+            foreach (var typeName in new[] { "byte", "ushort", "uint", "ulong" })
+            {
+                var source = @"
+using System;
+enum E : " + typeName + @"
+{
+    Zero,
+    One,
+    Two,
+    Three,
+    Four,
+    Five,
+    Six,
+    Seven
+}
+class C
+{
+    static void Main()
+    {
+        Console.WriteLine(M(E.Zero));
+        Console.WriteLine(M(E.One));
+        Console.WriteLine(M(E.Two));
+        Console.WriteLine(M(E.Three));
+        Console.WriteLine(M(E.Four));
+        Console.WriteLine(M(E.Five));
+        Console.WriteLine(M(E.Six));
+        Console.WriteLine(M(E.Seven));
+        Console.WriteLine(M((E)100));
+    }
+    static int M(E c) => c switch
+    {
+        > E.Five => 1,
+        E.Four => 2,
+        E.Three => 3,
+        E.Two => 4,
+        E.One => 5,
+        E.Zero => 6,
+        _ => 7, // handles E.Five
+    };
+}";
+                var expectedOutput = @"6
+5
+4
+3
+2
+7
+1
+1
+1";
+                var compilation = CreateCompilation(source, options: TestOptions.DebugExe, parseOptions: TestOptions.Regular.WithLanguageVersion(LanguageVersion.Preview));
+                compilation.VerifyDiagnostics(
+                    );
+                var comp = CompileAndVerify(compilation, expectedOutput: expectedOutput);
+            }
+        }
+
+        [Fact]
+        public void Relational_SignedEnumExhaustive()
+        {
+            foreach (var typeName in new[] { "byte", "ushort", "uint", "ulong" })
+            {
+                foreach (var withExhaustive in new[] { false, true })
+                {
+                    var source = @"
+enum E : " + typeName + @"
+{
+    Zero,
+    One,
+    Two,
+    Three,
+    Four,
+    Five,
+    Six,
+    Seven
+}
+class C
+{
+    static int M(E c) => c switch
+    {
+        > E.Five => 1,
+        E.Four => 2,
+        E.Three => 3,
+        E.Two => 4,
+        E.One => 5,
+        E.Zero => 6,
+" +
+(withExhaustive ? @"        E.Five => 7, // exhaustive
+" : "")
++ @"    };
+}";
+                    var compilation = CreateCompilation(source, parseOptions: TestOptions.Regular.WithLanguageVersion(LanguageVersion.Preview));
+                    if (withExhaustive)
+                    {
+                        compilation.VerifyDiagnostics(
+                            );
+                    }
+                    else
+                    {
+                        compilation.VerifyDiagnostics(
+                            // (15,28): warning CS8509: The switch expression does not handle all possible values of its input type (it is not exhaustive).
+                            //     static int M(E c) => c switch
+                            Diagnostic(ErrorCode.WRN_SwitchExpressionNotExhaustive, "switch").WithLocation(15, 28)
+                            );
+                    }
+                }
+            }
+        }
+
+        [Fact]
+        public void Relational_UnsignedEnumExhaustive()
```

Comment:
>Unsigned [](start = 31, length = 8)

"Signed"

Reply:
Will address in the next PR.

---
In reply to: [383373295](https://github.com/dotnet/roslyn/pull/41503#discussion_r383373295) [](ancestors = 383373295)